project-base/storefront/components/Basic/CopyTextBlock/CopyTextBlock.tsx

```diff
@@ -0,0 +1,36 @@
+import { useState } from 'react';
+
+type CopyTextBlockProps = {
+    textToCopy: string;
+    showCopyTextButton?: boolean;
+};
+
+export const CopyTextBlock: FC<CopyTextBlockProps> = ({ textToCopy, showCopyTextButton = true }) => {
+    const [isCopySuccess, setIsCopySuccess] = useState(false);
+
+    const handleCopy = async () => {
+        try {
+            await navigator.clipboard.writeText(textToCopy);
+            setIsCopySuccess(true);
+        } catch (err) {
+            setIsCopySuccess(false);
+        }
+    };
+
+    return (
+        <div className="rounded bg-greyLighter">
+            <div className="flex justify-end rounded bg-greyDark px-3 py-2">
+                {showCopyTextButton && (
+                    <button className="text-white" onClick={handleCopy}>
+                        {isCopySuccess ? 'Copied' : 'Copy Text'}
```

Comment:
Hmm, in code above I see that this `isCopySuccess` is supposed to be used for error handling but here is used for differentiate between state if used have copied or not. 

If promise will fall into catch user is not notified and he do infinite number of click without getting info what is wrong. That's why I would suggest to rather have three states `undefined`, `true` and `false` and decide what to display according to those (copied, copy text, error while copy). Or think about different way of handling like display toast message and don't change button text.

Reply:
To make it simple, I'll just change the text 👍